src/processors/default/addMimeTypeToProcessedTracks.ts

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash';
+
+import { Table } from '../../db/db';
+import { MimeEnum } from '../../types/media';
+import { NFT } from '../../types/nft';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+export enum SourceIPFS {
+  AUDIO = 'Audio',
+  ARTWORK = 'Artwork'
+}
+
+export type TrackNftJoin = ProcessedTrack & Partial<NFT>;
+
+const TIMEOUT = parseInt(process.env.METADATA_REQUEST_TIMEOUT!)
+const QUERY_LIMIT = process.env.QUERY_TRIGGER_BATCH_SIZE
+
+const missingMimeType: (source: SourceIPFS) => Trigger<undefined> = (source) => {
+  return async (clients) => {
+    // find approved tracks (and the nftID) which are missing a mimeType; skipping any which have had errors
+    const query = `select p.*, n.id as "nftId"
```

Comment:
hmm not sure it's a good idea to use the nft ID here - especially since:
 - there are usually 20+ nfts per track so with this outer join, there could be 25+ TrackNftJoin items for each track and so we'd repeat this 25 times
 - feels bad to use the nft process errors table for this, since again we'd be storing duplicate errors for every track and need to worry about the association between tracks and nfts just adding unnecessary complexity to this code

I do imagine in future we'd potentially want to check mimetype on nfts directly, rather that on tracks, but even then i don't think involving a relation here is good design.

I see the intention here is to use existing nft errors table, but feels over complex/messy and suggest we implement it with a cleaner trigger, so some options:
 a - Create a new processTrackErrors table for this
 b - Improve the nftErrors table to be more generic and support errors across different tables/data types
 c - Use the ipfsFiles table (which also has error fields in it) for this.

(c) looks easiest/simplest for now, and most appropriate since most the logic in this PR is actually ipfsFile specific, nevermind even track specific. I think implementing this to be ipfsFile specific rather than just track will also serve us well in future _(eg: if we want to process ipfsFiles on nfts or potentially videos and other things in future)_

So an idea to with this would be:
 - Actually store mimeType per file on the ipfsFiles
 - Similar trigger to those in `src/triggers/ipfs.ts` that triggers on records with null error and null mimeType
 - Processor stores both mimeType and errors on the ipfs file record, similar to how existing ipfsMediaUploader processor works
 - Above is great, now we have mimetypes on ipfsFiles, but still need to link them to tracks, so for that:
    - maybe just update the processedTracks view to join on ipfsFiles and include the mimetypes as part of the view and filter out null mimetypes in the view